### DISCUSSION TITLE
How to draw the ascendant ?

### DISCUSSION THREAD

**Discussion by sbalsom** [AstroDraw/AstroChart#31] [Q&A]
In settings.js, there is a case for drawing the Ascendant and Descendant symbols on the chart, so I know it can be done. However, there is no info on how to pass any value for the Asc/Desc point on the chart. How to do this ?

**Comment by ilyai** [comment]
Asc is an alias for cusp 1 and Desc is cusp 7, so you just pass them with the `cusps` parameter.

**Reply by egorrrrrrrrr** [reply]
It may be not correct, cause sometimes Asc not reflect cusp 1. For example you can calc Swiss ephemerides with _[swisseph](https://www.astro.com/ftp/swisseph/ ) for example for the Moscow at 2000-01-01 12:00 AM with Whole Equal house system and you will get

`{"planets":{"Sun":[254.9953899],"Moon":[191.0426103],"Mercury":[246.1812428],"Venus":[216.0739597],"Mars":[302.7421092],"Jupiter":[0.4918662],"Saturn":[15.672079],"Uranus":[290.0414028],"Neptune":[278.4344166],"Pluto":[226.6963517]},"cusps":[150,180,210,240,270,300,330,0,30,60,90,120],"points":{"As":[156.5202692]}}`

Here you have 150 degrees for cusp 1 and As is almost 157 degrees (greater than 150 for 7 degrees). I think in this situation it would be better to program additional function to set points As Mc (Dc Ic).

**Reply by ilyai** [reply]
In this case Asc should be 150 degrees as well. Displaying Asc and Cusp 1 at different degrees doesn't make astrological sense.

**Reply by andandio** [reply]
@egorrrrrrrrr is correct. When using a house system like Campanus House System, the first house cusp and the ascendant align. When using Whole or Equal house systems, 1st house/ 10th cusp will always be 0 degrees of the given sign, but Asc or Midheaven will not always be 0 degrees.  +1 to adding the abillity to optionally pass specific positions to the angles.

**Reply by roabhi** [reply]
Yes, I would like this feature as well please. As @andandio and @egorrrrrrrrr say some house systems set the first house as the place where the asc falls into but the cusp of that sign is always at 0º.  Not being able to draw angles separately from cusps makes the library useless for those systems.

**Reply by andandio** [reply]
@egorrrrrrrrr @roabhi I found a workaround for this, but it only works to draw the angles inside the chart, rather than external to the wheel as they normally occur with this library. You can add the angles to the planets array parameter when you draw the wheel like this:
```
var radix = chart
    .radix({ planets: allBodies, cusps: cusps })
  ```
  
  So, `allBodies` would include calculations for Asc, MC, etc.

**Reply by roabhi** [reply]
> @egorrrrrrrrr @roabhi I found a workaround for this, but it only works to draw the angles inside the chart, rather than external to the wheel as they normally occur with this library. You can add the angles to the planets array parameter when you draw the wheel like this:
> 
> ```
> var radix = chart
>     .radix({ planets: allBodies, cusps: cusps })
> ```
> 
> So, `allBodies` would include calculations for Asc, MC, etc.

Hey thanks for sharing. Will give it a try.  DSC and IC also need to be included right?

**Reply by andandio** [reply]
If you want them to show up, then yes. I did not include them as I think (?) they are always inverse of ASC & MC. 

I also used `CUSTOM_SYMBOL_FN` in settings to pass a custom svg/text for them inside the wheel, and then just css to hide the text that still gets drawn for the angles external to the wheel.

**Reply by roabhi** [reply]
Thanks!, How to use `CUSTOM_SYMBOL_FN` I am just getting started with the library...
